### PR TITLE
batch-submitter: updated config

### PR DESCRIPTION
**Pull request by tynes**
**Description**
Updates the config in a backwards compatible way so that it works with `bcfg`. Now the prefix `BATCH_SUBMITTER_<config-name>` will be parsed as config as well as CLI arguments. This was done in a backwards compatible way such that the previous env vars are parsed as the default values. We can move towards phasing out the old configuration style in the future.

Also uses the `USE_SENTRY` config option to turn on sentry

### CODE REVIEW

File: packages/batch-submitter/src/exec/run-batch-submitter.ts [hunk 1]
```diff
@@ -112,50 +73,195 @@ const requiredEnvVars: RequiredEnvVars = {
...
+  const sentryTraceRate = config.ufloat(
+    'sentry-trace-rate',
+    parseInt(env.SENTRY_TRACE_RATE, 10) || 0.05
+  )
+
+  /* Logger */
+  const name = 'oe:batch_submitter:init'
+  let logger
+
+  if (config.bool('use-sentry')) {
```
Comment (annieke) [inline packages/batch-submitter/src/exec/run-batch-submitter.ts:101]: to confirm, this is the only new variable added / changed in this PR?

**Review by annieke — CHANGES_REQUESTED** [review]
this change overall looks good to me, can we update `.env.example` to the new format?

File: packages/batch-submitter/src/exec/run-batch-submitter.ts [hunk 2]
```diff
@@ -4,8 +4,17 @@ import { Logger, Metrics } from '@eth-optimism/common-ts'
 import { exit } from 'process'
 import { Signer, Wallet } from 'ethers'
 import { JsonRpcProvider, TransactionReceipt } from '@ethersproject/providers'
-import { config } from 'dotenv'
-config()
+import * as dotenv from 'dotenv'
+import Config from 'bcfg'
+
+interface Bcfg {
```
Comment (smartcontracts) [inline packages/batch-submitter/src/exec/run-batch-submitter.ts:10]: I think we have this in multiple places now. We could probably consider consolidating this into a single interface in `core-utils` or even make a PR against https://github.com/DefinitelyTyped/DefinitelyTyped adding typedefs for bcfg.

Comment (tynes) [reply packages/batch-submitter/src/exec/run-batch-submitter.ts:10]: I'm happy to move the interface into `core-utils` for now and fleshing out a full one against definitelytyped in the future

**Comment by tynes** [comment]
> this change overall looks good to me, can we update .env.example to the new format?

This is backwards compatible, with updating `.env.example` to the new format I'd like to also update the `ops` directory as well as our `k8s` deployments and then remove the need to be backwards compatible

File: packages/batch-submitter/src/exec/run-batch-submitter.ts (hunk 1 above)
Comment (tynes) [reply packages/batch-submitter/src/exec/run-batch-submitter.ts:101]: Yes this is the only new variable added - it is used to be explicit and we should make sure that we use the same name across the different services

**Comment by tynes** [comment]
> > this change overall looks good to me, can we update .env.example to the new format?
> 
> This is backwards compatible, with updating `.env.example` to the new format I'd like to also update the `ops` directory as well as our `k8s` deployments and then remove the need to be backwards compatible

To follow up @annieke, I do not plan on updating the example env until the backwards compatibility is removed and we fully use the scheme of prefixed env vars. Right now the example env is acting as a forcing function for backwards compatibility

File: packages/batch-submitter/src/exec/run-batch-submitter.ts (hunk 1 above)
Comment (annieke) [reply packages/batch-submitter/src/exec/run-batch-submitter.ts:92]: @tynes this defaults to false then i'm assuming? would appreciate adding this to `.env.example`!

**Review by annieke — APPROVED** [review]
lgtm with one nit

**Comment by tynes** [comment]
I've also added a changeset for the other impacted packages

File: packages/batch-submitter/src/exec/run-batch-submitter.ts [hunk 3]
```diff
@@ -112,50 +64,195 @@ const requiredEnvVars: RequiredEnvVars = {
...
+
+  // Parse config
+  const env = process.env
+  const environment = config.str('node-env', env.NODE_ENV)
+  const network = config.str('eth-network-name', env.ETH_NETWORK_NAME)
+  const release = `batch-submitter@${env.npm_package_version}`
+  const sentryDsn = config.str('sentry-dsn', env.SENTRY_DSN)
+  const sentryTraceRate = config.ufloat(
+    'sentry-trace-rate',
+    parseInt(env.SENTRY_TRACE_RATE, 10) || 0.05
```
Comment (tynes) [inline packages/batch-submitter/src/exec/run-batch-submitter.ts:85]: This should be a `parseFloat`

**Pull request merged**